source/math/FloatVectorList.ooc

```diff
@@ -247,15 +235,93 @@ FloatVectorList: class extends VectorList<Float> {
 		tempVector free()
 		result
 	}
+	/* calculate median without copying and sorting the vector
+			WARNING: this function can partially rearange the elements in vector
+	*/
+	fastMedian: func -> Float {
+		data := this _vector _backend as Float*
+		result := This _nthElement(this _vector _backend as Float*, 0, this count - 1, this count / 2)
+		if (Int even(this count))
+			result = (result + This _nthElement(this _vector _backend as Float*, 0, this count - 1, this count / 2 - 1)) / 2
+		result
+	}
 	movingMedianFilter: func (windowSize: Int) -> This {
 		result := This new(this count)
 		start := -((windowSize - 1) / 2)
 		for (i in 0..this count) {
 			range := (start..(start + windowSize - 1)) + i
 			elementsInWindow := this getSlice(range clamp(0, this count-1))
-			result add((elementsInWindow as This) median())
+			result add((elementsInWindow as This) fastMedian())
```

Comment:
A `median` function that takes a range (start and stop indexes) instead of a temporary `VectorList` would improve performance a lot, here, even for tiny `VectorList`s. We create a new `VectorList` called `elementsInWindow` `this count` times and free it immediately after computing its median.